src/slurm_plugin/slurm_resources.py

```diff
@@ -320,6 +324,8 @@ def is_state_healthy(self, terminate_drain_nodes, terminate_down_nodes, log_warn
         """Check if a slurm node's scheduler state is considered healthy."""
         # Check to see if node is in DRAINED, ignoring any node currently being replaced
         if self.is_drained() and terminate_drain_nodes:
+            if self.is_reboot_issued() or self.is_reboot_requested() or self.is_power_down():
```

Comment:
Can we handle the entire reboot logic (`self.is_reboot_issued() or self.is_reboot_requested()`) as a separate block at the beginning of the is_state_healthy, eventually by moving it to a separate function which also writes a log line when the condition is valid?Otherwise this function is becoming harder and harder to read.

Also, what happens if the node fails to reboot? do we detect this case? is slurm changing the state for the node after a timeout and we eventually mark the node as unhealthy?

Reply:
> Can we handle the entire reboot logic (`self.is_reboot_issued() or self.is_reboot_requested()`) as a separate block at the beginning of the is_state_healthy, eventually by moving it to a separate function which also writes a log line when the condition is valid?Otherwise this function is becoming harder and harder to read.

Here are some considerations on my side:
* the way the checks on the reboot flags are nested with the checks on the DRAINED / DOWN statuses reflects Slurm's workflow in setting the reboot flags after a reboot request. Moving those checks out of the `if` statements for DRAINED/DOWN means losing that structure.
* I generally agree with you that the `is_state_healthy` method risks becoming more and more unreadable, but I think it's out of scope: we would something like `handle_reboot_states()`, `handle_drained_state()`, `handle_down_state()`, and this should be done in a homogeneous manner throughout `is_state_healthy()`. And I'm not sure yet this is the best way to deal with this.

> Also, what happens if the node fails to reboot? do we detect this case? is slurm changing the state for the node after a timeout and we eventually mark the node as unhealthy?

Correct. After `ResumeTimeout`, if the node hasn't responded yet it will be marked as DOWN and later `clustermgtd` will treat it as unhealthy.